pkg/ctl/create/utils.go

```diff
@@ -0,0 +1,230 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gobwas/glob"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/weaveworks/eksctl/pkg/ami"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha4"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+	"github.com/weaveworks/eksctl/pkg/eks"
+)
+
+// NodeGroupFilter holds filter configuration
+type NodeGroupFilter struct {
+	IgnoreAllExisting bool
+
+	existing sets.String
+	only     []glob.Glob
+	onlySpec string
+}
+
+// NewNodeGroupFilter create new NodeGroupFilter instance
+func NewNodeGroupFilter() *NodeGroupFilter {
+	return &NodeGroupFilter{
+		IgnoreAllExisting: true,
+
+		existing: sets.NewString(),
+	}
+}
+
+// ApplyOnlyFilter parses given globs for exclusive filtering
+func (f *NodeGroupFilter) ApplyOnlyFilter(globs []string, cfg *api.ClusterConfig) error {
+	for _, gs := range globs {
+		g, err := glob.Compile(gs)
+		if err != nil {
+			return errors.Wrapf(err, "parsing glob filter %q", gs)
+		}
+		f.only = append(f.only, g)
+	}
+	f.onlySpec = strings.Join(globs, ",")
+	return f.onlyFilterMatchesAnything(cfg)
+}
+
+func (f *NodeGroupFilter) onlyFilterMatchesAnything(cfg *api.ClusterConfig) error {
+	if len(f.only) == 0 {
+		return nil
+	}
+	for i := range cfg.NodeGroups {
+		for _, g := range f.only {
+			if g.Match(cfg.NodeGroups[i].Name) {
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("no nodegroups match filter specification: %q", f.onlySpec)
+}
+
+// GetExisting uses stackManager to list existing nodegroup stacks and configures
+// the filter accordingly
+func (f *NodeGroupFilter) GetExisting(stackManager *manager.StackCollection) error {
```

Comment:
Nit: Any time I see a function or method named "Get<xxx>" it makes me expect it to return some fetched data, rather than perform initialization

Reply:
Yeah, I was thinking to rename this... perhaps `ApplyExistingFilter` (so it matches the `ApplyOnlyFilter`)?